templates/repo/code/recently_pushed_new_branches.tmpl

```diff
@@ -1,10 +1,22 @@
 {{range .RecentlyPushedNewBranches}}
+	{{$pullRepo := $.Repository}}
+	{{$baseRepo := $.Repository}}
+	{{$branchName := .Name}}
+	{{if .Repo}}
+		{{$pullRepo = .Repo}}
+		{{if and $.Repository.IsFork}}
+			{{$baseRepo = $.Repository.BaseRepo}}
+		{{end}}
+		{{if not (eq .Repo.ID $.Repository.ID)}}
+			{{$branchName = (print .Repo.FullName ":" .Name)}}
+		{{end}}
+	{{end}}
 	<div class="ui positive message gt-df gt-ac">
 		<div class="gt-f1">
 			{{$timeSince := TimeSince .UpdatedUnix.AsTime $.locale}}
-			{{$.locale.Tr "repo.pulls.recently_pushed_new_branches" (PathEscapeSegments .Name) $timeSince | Safe}}
+			{{$.locale.Tr "repo.pulls.recently_pushed_new_branches" (PathEscapeSegments $branchName) $timeSince | Safe}}
```

Comment:
```
{{$.locale.Tr "repo.pulls.recently_pushed_new_branches" (PathEscapeSegments $branchName) $timeSince | Safe}}
```			
`PathEscapeSegments` This will create the problem when `branchname` contains `#` . like `branchname#123`

Double encoding of html Escape is going on.

There was an issue raised 
Screenshot 1:
https://user-images.githubusercontent.com/23124682/252675332-4f8da1bf-40f6-4442-bf39-a261bd7289fa.png

Screenshot 2:
https://user-images.githubusercontent.com/23124682/252675084-e3bc020c-1cb8-4801-87fe-dd79de9bcd02.png

### Issue link: -
https://github.com/go-gitea/gitea/issues/25830

@yp05327  FYI.

Reply:
Not only here, for example: #25864
I think we should do this in another PR.

Reply:
When I try to test this, I notice that we have more problems:
https://try.gitea.io/yp05327/testrepo/src/branch/main
If you select branch `test#11`, the url will be 
https://try.gitea.io/yp05327/testrepo/src/branch/test#11
but in the list it is `test`
![image](https://github.com/go-gitea/gitea/assets/18380374/8df66a64-6fd4-4adb-b3bd-f8cc9620ff87)
Maybe we need a summary. 🤔

Reply:
@puni9869 
I created a new issue to discuss branch name problems: #25865
It seems that it is a global problem, so there's no plan to fix it in this PR now.

Reply:
> When I try to test this, I notice that we have more problems: https://try.gitea.io/yp05327/testrepo/src/branch/main If you select branch `test#11`, the url will be https://try.gitea.io/yp05327/testrepo/src/branch/test#11 but in the list it is `test` ![image](https://user-images.githubusercontent.com/18380374/253198968-8df66a64-6fd4-4adb-b3bd-f8cc9620ff87.png) Maybe we need a summary. 🤔


Thanks yp05327  for the quick response
Here is the url of `test#11`.See its encoded  `test%2311`.

https://try.gitea.io/yp05327/testrepo/src/branch/test%2311

PS: I have checked be visiting `branches` tab 
<img width="837" alt="image" src="https://github.com/go-gitea/gitea/assets/80308335/ed5fc437-92cd-477d-9f5f-b0c627c229f6">

That was the problem I was pointing in my comments.

`test` --> when you visite `https://try.gitea.io/yp05327/testrepo/src/branch/test#11` ie`test#11` it will redirect to  `test` branch
<img width="645" alt="image" src="https://github.com/go-gitea/gitea/assets/80308335/d66724de-0460-4b93-9e27-5413ad8fa89d">
The `#` in the `test#11` will be treated as `hash routing`.

`test#11` ---> when you visite `https://try.gitea.io/yp05327/testrepo/src/branch/test%2311` ie `test%2311` you will land to  `test#11`

Reply:
I appreciate your feedback and Opening a new issue on specific to branch name.